src/ImageSharp/Formats/Png/PngEncoderCore.cs

```diff
@@ -613,6 +614,21 @@ private void WritePhysicalChunk<TPixel>(Stream stream, Image<TPixel> image)
             }
         }
 
+        /// <summary>
+        /// Writes the eXIf chunk to the stream, if any EXIF Profile values are present in the meta data.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="stream">The <see cref="Stream"/> containing image data.</param>
+        /// <param name="image">The image.</param>
+        private void WriteExifChunk<TPixel>(Stream stream, Image<TPixel> image)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            if (image.MetaData.ExifProfile?.Values.Count > 0)
```

Comment:
We need to alter this slightly as there's extra checks and balances.

`image.MetaData.SyncProfiles();` is required before writing according to the jpeg encoder and we need additional length checks as the spec dictates it's length should match jpeg limitations. 

https://github.com/SixLabors/ImageSharp/blob/d171acba8051063a68467e912f1a81fae1126ed8/src/ImageSharp/Formats/Jpeg/JpegEncoderCore.cs#L608

Reply:
i have added the SyncProfile before writing the chunk. 

For the length check: I could add a check, if the length exceeds 2^16, like the JpegEncoder does, and throw an exception, but that feels wrong to me. This restriction is Jpeg specific not PNG specific. 

In the recommendations they state, when writing PNG Exif to Jpeg: 
> the total length of the eXIf chunk data may need to be adjusted

But how to decide what to cut off here is not clear to me. I really would like to hear @dlemstra opinion on that.

Reply:
Looking at this it appears we should simply the split data into 64k chunks.  http://dev.exiv2.org/issues/1232

That means we need an extend method, like the ICC readeron decoding.
https://github.com/SixLabors/ImageSharp/blob/d7bd82b79ee396a42b13a83563e4fbdd951c2a55/src/ImageSharp/Formats/Jpeg/PdfJsPort/PdfJsJpegDecoderCore.cs#L508